### PR TITLE
Drop Python versions that are unsupported by core team

### PR DESCRIPTION
**Pull request by hynek**
Fixes: #375

Also fix combinatorial explosion of test envs in tox.ini.

The fixes to tox.ini take local test builds from ~15mins to ~2mins.

This is only a fix to build, metadata, and docs. I'd massage it with pyupgrade next to simplify the code, but I want to keep this small.

### CODE REVIEW

**Comment by hynek** [comment]
Clicking through into the listed codecov files and looking at the "Coverage Changes" tab they all say "Everything is accounted for! No changes detected that need to be reviewed."

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -67,12 +67,6 @@ jobs:
           - name: Windows
```
Comment (adiroiban) [inline .github/workflows/ci.yml:67]: Maybe also do

```suggestion
          - name: Win
```


To work around GitHub UI issues

image

**Review by adiroiban — APPROVED** [review]
Thanks! Only a minor comment that is not really related to this PR.

Looking forward to a bonanza of py3 only PR :)

**Comment by adiroiban** [comment]
Thanks!

**Comment by q0w** [comment]
tomli can be optional for py311+, no?

**Comment by adiroiban** [comment]
I don't think we support 3.11 yet. Support is only up to 3.10.

PR welcomed for 3.11 :)

**Comment by q0w** [comment]
what do you think about adding linting tools?

**Comment by hynek** [comment]
The work I’m currently investing in towncrier is mainly to improve contributor experience and simplify maintenance.

I’ll offer a PR that does the usual black/isort/pyuprade dance via pre-commit (within Tox as fallback) together with pre-commit.ci (runs super fast over PRs and fixes in place) later. I mean there’s already a black config in pyproject.toml. That should help a lot.

**Pull request merged**